server/src/main/kotlin/suwayomi/tachidesk/manga/impl/download/DownloadManager.kt

```diff
@@ -124,6 +138,31 @@ object DownloadManager {
         }
     }
 
+    private val saveQueueFlow = MutableStateFlow(Unit)
+    init {
+        val mutex = Mutex()
+        val saveQueue = Semaphore(1)
+
+        // this makes sure that only one save process is active at a time and only the latest emission causes a new save.
+        // example:
+        // emit A - start save
+        // emit B - wait for active save
+        // emit C - omit B - wait for active save
+        // emit D - omit C - wait for active save
+        // emit A - save done
+        // emit D - start save
+        // ...
+        saveQueueFlow.flatMapLatest {
+            saveQueue.acquire()
+            flow { emit(it) }
+        }.onEach {
+            mutex.withLock {
+                saveDownloadQueue()
+                saveQueue.release()
+            }
+        }.launchIn(scope)
+    }
```

Comment:
I'm pretty sure this can just be replaced by using

```
val saveQueueFlow = MutableSharedFlow(extraBufferCapacity = 1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
```
instead of this magic

Reply:
but everyone needs a little magic in their lives - updated
![your-a-wizard-harry](https://github.com/Suwayomi/Tachidesk-Server/assets/50052685/16fd1a4b-6be0-47f7-8b89-58e336baeace)